### ISSUE TITLE
Crash when providing 0 or extra large values for XCURSOR_SIZE

### ISSUE THREAD

**Issue by house-of-vanity** [alacritty/alacritty#3782]
Alacritty doesn't start on KDE (Wayland) desktop environment using native wayland feature.
It works with XWayland feature `WINIT_UNIX_BACKEND=x11 alacritty`

### System

OS: Arch Linux 5.6.14
Version: alacritty 0.4.2
Linux/BSD: Wayland, KDE Plasma 5.18.5

### Versions

```
[ab@ab alacritty]$ pacman -Qs wayland
local/kwayland 5.70.0-1 (kf5)
    Qt-style Client and Server library wrapper for the Wayland libraries
local/kwayland-integration 5.18.5-1 (plasma)
    Provides integration plugins for various KDE frameworks
local/lib32-libva1 1.8.3-3
    Video Acceleration (VA) API for Linux
local/lib32-wayland 1.18.0-2
    A computer display server protocol
local/libva 2.7.1-1
    Video Acceleration (VA) API for Linux
local/libva1 1.8.3-2
    Video Acceleration (VA) API for Linux
local/plasma-wayland-session 5.18.5-3
    Plasma Wayland session
local/wayland 1.18.0-2
    A computer display server protocol
local/wayland-protocols 1.20-1
    Specifications of extended Wayland protocols
local/weston 8.0.0-2
    Reference implementation of a Wayland compositor
local/xorg-server-xwayland 1.20.8-2 (xorg)
    run X clients under wayland
```

### Logs
**alacritty 0.4.2**
```
[ab@pc alacritty]$ RUST_BACKTRACE=full alacritty --version
alacritty 0.4.2

[ab@pc alacritty]$ RUST_BACKTRACE=full alacritty -vvv
Created log file at "/tmp/Alacritty-42727.log"
[2020-05-28 13:55:02.893181513] [INFO] Welcome to Alacritty
[2020-05-28 13:55:02.893219956] [INFO] No configuration file found
[2020-05-28 13:55:02.900903438] [DEBUG] Estimated DPR: 1
[2020-05-28 13:55:02.900915560] [DEBUG] Estimated Cell Size: 8 x 17
[2020-05-28 13:55:02.900923996] [DEBUG] Estimated Dimensions: None
thread 'main' panicked at 'Memory allocation failure while loading a theme.', /build/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/cursor.rs:80:5
stack backtrace:
   0:     0x557ad3ce668f - <unknown>
   1:     0x557ad3bb774c - <unknown>
   2:     0x557ad3ce5ba1 - <unknown>
   3:     0x557ad3ce5700 - <unknown>
   4:     0x557ad3ce4cdf - <unknown>
   5:     0x557ad3d0d6a4 - <unknown>
   6:     0x557ad3cb3535 - <unknown>
   7:     0x557ad3ac4149 - <unknown>
   8:     0x557ad3a940d6 - <unknown>
   9:     0x557ad3b3d158 - <unknown>
  10:     0x557ad3ad58b3 - <unknown>
  11:     0x557ad3b38963 - <unknown>
  12:     0x7f150b9a4002 - __libc_start_main
  13:     0x557ad39b90de - <unknown>
  14:                0x0 - <unknown>
```

**alacritty 0.5.0-dev (2b2b2bf)**
```
[ab@pc alacritty]$ RUST_BACKTRACE=full ./target/release/alacritty --version
alacritty 0.5.0-dev (2b2b2bf)

[ab@pussy alacritty]$ RUST_BACKTRACE=full ./target/release/alacritty -vvv
Created log file at "/tmp/Alacritty-44255.log"
[2020-05-28 13:58:17.550633231] [INFO] Welcome to Alacritty
[2020-05-28 13:58:17.550678166] [INFO] No configuration file found
[2020-05-28 13:58:17.558501110] [DEBUG] Estimated DPR: 1
[2020-05-28 13:58:17.558514165] [DEBUG] Estimated Cell Size: 8 x 17
[2020-05-28 13:58:17.558521408] [DEBUG] Estimated Dimensions: None
thread 'main' panicked at 'Memory allocation failure while loading a theme.', /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/cursor.rs:80:5
stack backtrace:
   0:     0x561c6c47359c - backtrace::backtrace::libunwind::trace::h90669f559fb267f0
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1:     0x561c6c47359c - backtrace::backtrace::trace_unsynchronized::hffde4e353d8f2f9a
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2:     0x561c6c47359c - std::sys_common::backtrace::_print_fmt::heaf44068b7eaaa6a
                               at src/libstd/sys_common/backtrace.rs:77
   3:     0x561c6c47359c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h88671019cf081de2
                               at src/libstd/sys_common/backtrace.rs:59
   4:     0x561c6c32061c - core::fmt::write::h4e6a29ee6319c9fd
                               at src/libcore/fmt/mod.rs:1052
   5:     0x561c6c472e36 - std::io::Write::write_fmt::hf06b1c86d898d7d6
                               at src/libstd/io/mod.rs:1426
   6:     0x561c6c472aa5 - std::sys_common::backtrace::_print::h404ff5f2b50cae09
                               at src/libstd/sys_common/backtrace.rs:62
   7:     0x561c6c472aa5 - std::sys_common::backtrace::print::hcc4377f1f882322e
                               at src/libstd/sys_common/backtrace.rs:49
   8:     0x561c6c472aa5 - std::panicking::default_hook::{{closure}}::hc172eff6f35b7f39
                               at src/libstd/panicking.rs:204
   9:     0x561c6c472229 - std::panicking::default_hook::h7a68887d113f8029
                               at src/libstd/panicking.rs:224
  10:     0x561c6c472229 - std::panicking::rust_panic_with_hook::hb7ad5693188bdb00
                               at src/libstd/panicking.rs:472
  11:     0x561c6c499664 - std::panicking::begin_panic::heff5ded1ce9035cb
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panicking.rs:399
  12:     0x561c6c42c1d6 - smithay_client_toolkit::pointer::theme::ScaledThemeList::get_cursor::hd53fdf28c8bd31cb
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/cursor.rs:80
  13:     0x561c6c42c1d6 - smithay_client_toolkit::pointer::theme::ThemedPointer::set_cursor_with_scale::h2368add9f8b55b6b
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/smithay-client-toolkit-0.6.6/src/pointer/theme.rs:215
  14:     0x561c6c21c7aa - winit::platform_impl::platform::wayland::window::Window::set_cursor_icon::h2d45996ff81cd08a
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/smithay-client-toolkit-0.6.6/src/pointer/mod.rs:136
  15:     0x561c6c21c7aa - winit::platform_impl::platform::Window::set_cursor_icon::hd4ed4443268d1b7f
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.22.2/src/platform_impl/linux/mod.rs:326
  16:     0x561c6c21c7aa - winit::window::Window::set_cursor_icon::h3b5ee8683a95e4f1
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.22.2/src/window.rs:698
  17:     0x561c6c21c7aa - alacritty::window::Window::new::hf8a24080303db192
                               at alacritty/src/window.rs:190
  18:     0x561c6c1f1225 - alacritty::display::Display::new::h6d647766e0934371
                               at alacritty/src/display.rs:166
  19:     0x561c6c178865 - alacritty::run::h90f7d2d4fc169c7f
                               at alacritty/src/main.rs:148
  20:     0x561c6c178865 - alacritty::main::h1ac57b13d02f1e1a
                               at alacritty/src/main.rs:115
  21:     0x561c6c271ca3 - std::rt::lang_start::{{closure}}::hbe93053897a51944
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  22:     0x561c6c1738c5 - main
  23:     0x7fec82f1b002 - __libc_start_main
  24:     0x561c6c0f519e - _start
  25:                0x0 - <unknown>

```

**Comment by kchibisov** [comment]
It really seems like not our bug tbqh. The bug is that `libwayland-cursor` is failing to allocate memory, could you give us more details about your setup? Is it some sandboxing, etc? Does running other apps using libwayland-cursor works for you?

Also, if you can run `cargo run --example kbd_input` from https://github.com/smithay/client-toolkit it'll be nice to hear whether master crashes for you and whether the same example crashes for you on https://github.com/Smithay/client-toolkit/tree/v0.6.x.

**Comment by house-of-vanity** [comment]
> could you give us more details about your setup, Is it some sandboxing, etc? 
I'm using common home PC, it has enough RAM (16GiB) and AMD Ryzen 5 3600. Linux setup is quite straightforward without any mandatory access control system enabled and kernel hardening settings which could interfere execution. Also I've performed full system upgrade.

> Does running other apps using libwayland-cursor works for you
I'm not sure that i know how to test this.

**master branch** 
kbd_input example handle kbd input well but panicked if mouse move over window.
```
    Updating crates.io index
   Compiling smithay-client-toolkit v0.9.1 (/home/ab/repos/client-toolkit)
    Finished dev [unoptimized + debuginfo] target(s) in 3.15s
     Running `target/debug/examples/kbd_input`
Window states: []
Window states: []
Gained focus on seat '' while 0 keys pressed.
Modifiers changed to ModifiersState { ctrl: false, alt: false, shift: false, caps_lock: false, logo: false, num_lock: true } on seat ''.
Window states: [Activated]
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2791:10
stack backtrace:
   0:     0x559719371a94 - backtrace::backtrace::libunwind::trace::h90669f559fb267f0
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1:     0x559719371a94 - backtrace::backtrace::trace_unsynchronized::hffde4e353d8f2f9a
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2:     0x559719371a94 - std::sys_common::backtrace::_print_fmt::heaf44068b7eaaa6a
                               at src/libstd/sys_common/backtrace.rs:77
   3:     0x559719371a94 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h88671019cf081de2
                               at src/libstd/sys_common/backtrace.rs:59
   4:     0x55971938b40c - core::fmt::write::h4e6a29ee6319c9fd
                               at src/libcore/fmt/mod.rs:1052
   5:     0x55971936f007 - std::io::Write::write_fmt::hf06b1c86d898d7d6
                               at src/libstd/io/mod.rs:1426
   6:     0x559719373c35 - std::sys_common::backtrace::_print::h404ff5f2b50cae09
                               at src/libstd/sys_common/backtrace.rs:62
   7:     0x559719373c35 - std::sys_common::backtrace::print::hcc4377f1f882322e
                               at src/libstd/sys_common/backtrace.rs:49
   8:     0x559719373c35 - std::panicking::default_hook::{{closure}}::hc172eff6f35b7f39
                               at src/libstd/panicking.rs:204
   9:     0x559719373921 - std::panicking::default_hook::h7a68887d113f8029
                               at src/libstd/panicking.rs:224
  10:     0x55971937429a - std::panicking::rust_panic_with_hook::hb7ad5693188bdb00
                               at src/libstd/panicking.rs:472
  11:     0x559719373e80 - rust_begin_unwind
                               at src/libstd/panicking.rs:380
  12:     0x55971938a8d1 - core::panicking::panic_fmt::hb1f3e14b86a3520c
                               at src/libcore/panicking.rs:85
  13:     0x55971938a895 - core::panicking::panic_bounds_check::h09b793daa6d169ff
                               at src/libcore/panicking.rs:63
  14:     0x55971921d071 - <usize as core::slice::SliceIndex<[T]>>::index::hb0683ca0eb1bbd13
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2791
  15:     0x559719219328 - core::slice::<impl core::ops::index::Index<I> for [T]>::index::h0b25e12589977420
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2656
  16:     0x55971921eb9b - <alloc::vec::Vec<T> as core::ops::index::Index<I>>::index::h37f91e6910c39b89
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/vec.rs:1883
  17:     0x55971921ab67 - <wayland_cursor::Cursor as core::ops::index::Index<usize>>::index::h4757a71fc93afbb5
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-cursor-0.26.6/src/lib.rs:242
  18:     0x559719109046 - smithay_client_toolkit::seat::pointer::theme::PointerInner::update_cursor::h143f9ee80ab26cad
                               at src/seat/pointer/theme.rs:206
  19:     0x559719109641 - smithay_client_toolkit::seat::pointer::theme::ThemedPointer::set_cursor::h86641305b5ed0e35
                               at src/seat/pointer/theme.rs:258
  20:     0x559719034660 - smithay_client_toolkit::window::concept_frame::change_pointer::h1b09d1ff377b607e
                               at src/window/concept_frame.rs:921
  21:     0x55971904b28b - <smithay_client_toolkit::window::concept_frame::ConceptFrame as smithay_client_toolkit::window::Frame>::new_seat::{{closure}}::h254387aa75e38197
                               at src/window/concept_frame.rs:412
  22:     0x55971909448c - smithay_client_toolkit::seat::pointer::theme::ThemeManager::theme_pointer_with_impl::{{closure}}::h5dec6631955ebc10
                               at src/seat/pointer/theme.rs:117
  23:     0x5597190b4196 - wayland_client::proxy::Main<I>::quick_assign::{{closure}}::h1dac5f8cac4365c8
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/proxy.rs:283
  24:     0x55971911a173 - wayland_commons::filter::Filter<E>::send::h79d55288497500e6
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-commons-0.26.6/src/filter.rs:95
  25:     0x55971905ce76 - wayland_client::imp::proxy::ProxyInner::assign::{{closure}}::h4441b0a7a06d50d1
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/native_lib/proxy.rs:251
  26:     0x5597192fad06 - <alloc::boxed::Box<F> as core::ops::function::Fn<A>>::call::heaa6f61590a7f87e
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/boxed.rs:1029
  27:     0x5597192bcc73 - wayland_client::imp::proxy::proxy_dispatcher::{{closure}}::{{closure}}::h75dc2c04966e47b8
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/native_lib/proxy.rs:421
  28:     0x5597192fe1f3 - scoped_tls::ScopedKey<T>::with::he6c0b69f0c3a1352
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:171
  29:     0x5597192c0200 - wayland_client::imp::proxy::proxy_dispatcher::{{closure}}::h071b463c5c31dbf3
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/native_lib/proxy.rs:418
  30:     0x5597192dc358 - std::panicking::try::do_call::h7c82277dc45f9e01
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panicking.rs:305
  31:     0x5597193779d7 - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:86
  32:     0x5597192dbcdb - std::panicking::try::hddd5470cd3410173
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panicking.rs:281
  33:     0x5597192f5781 - std::panic::catch_unwind::ha9e5de4f4b750679
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panic.rs:394
  34:     0x5597192baffa - wayland_client::imp::proxy::proxy_dispatcher::hd629ec981173e977
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/native_lib/proxy.rs:396
  35:     0x7f858809ef6b - <unknown>
  36:     0x7f858809b662 - <unknown>
  37:     0x7f858809cccc - wl_display_dispatch_queue_pending
  38:     0x559718fa61f7 - wayland_client::imp::event_queue::EventQueueInner::dispatch_pending::{{closure}}::hb5393a42e02c7606
                               at /home/ab/repos/client-toolkit/<::wayland_sys::ffi_dispatch macros>:2
  39:     0x559718fa65fb - wayland_client::imp::event_queue::with_dispatch_meta::{{closure}}::h4e0edd2920d07787
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/native_lib/event_queue.rs:24
  40:     0x55971901ea2f - scoped_tls::ScopedKey<T>::set::h9a94166e332e109c
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137
  41:     0x559718fa6336 - wayland_client::imp::event_queue::with_dispatch_meta::h0cfd46400d4708f7
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/native_lib/event_queue.rs:24
  42:     0x559718fa6149 - wayland_client::imp::event_queue::EventQueueInner::dispatch_pending::h8a4d6a36251e79ee
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/native_lib/event_queue.rs:62
  43:     0x559718fd687e - wayland_client::event_queue::EventQueue::dispatch_pending::h26ddc6be24f3f740
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.26.6/src/event_queue.rs:162
  44:     0x559718fadb6a - smithay_client_toolkit::event_loop::WaylandSource::quick_insert::{{closure}}::h5229dd700f8a47c5
                               at /home/ab/repos/client-toolkit/src/event_loop.rs:41
  45:     0x559718f9f557 - <core::cell::RefCell<calloop::sources::Dispatcher<Data,S,F>> as calloop::sources::EventDispatcher<Data>>::process_events::{{closure}}::h7ed91e0c13289f58
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/calloop-0.6.2/src/sources/mod.rs:123
  46:     0x559718fae3c0 - <smithay_client_toolkit::event_loop::WaylandSource as calloop::sources::EventSource>::process_events::h7f17b81dfa17cb3c
                               at /home/ab/repos/client-toolkit/src/event_loop.rs:81
  47:     0x559718fbf295 - <core::cell::RefCell<calloop::sources::Dispatcher<Data,S,F>> as calloop::sources::EventDispatcher<Data>>::process_events::h0b8c9f3c5a45f03f
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/calloop-0.6.2/src/sources/mod.rs:123
  48:     0x559718ff5c93 - calloop::loop_logic::EventLoop<Data>::dispatch_events::h83318d36a1dd7455
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/calloop-0.6.2/src/loop_logic.rs:268
  49:     0x559718ff6fb1 - calloop::loop_logic::EventLoop<Data>::dispatch::hf4c1bc7f8a5ce7cf
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/calloop-0.6.2/src/loop_logic.rs:312
  50:     0x559719008c94 - kbd_input::main::h24d28ed69819d952
                               at examples/kbd_input.rs:185
  51:     0x559719007ae0 - std::rt::lang_start::{{closure}}::hd8c5cc347d9b3911
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  52:     0x559719373d63 - std::rt::lang_start_internal::{{closure}}::hb26e39676675046f
                               at src/libstd/rt.rs:52
  53:     0x559719373d63 - std::panicking::try::do_call::he4701ab6e48d80c0
                               at src/libstd/panicking.rs:305
  54:     0x5597193779d7 - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:86
  55:     0x559719374740 - std::panicking::try::hd3de25f3cb7024b8
                               at src/libstd/panicking.rs:281
  56:     0x559719374740 - std::panic::catch_unwind::h86c02743a24e3d92
                               at src/libstd/panic.rs:394
  57:     0x559719374740 - std::rt::lang_start_internal::h9cf8802361ad86c2
                               at src/libstd/rt.rs:51
  58:     0x559719007ab9 - std::rt::lang_start::hb2f42dc3a80646ec
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  59:     0x55971900bd0a - main
  60:     0x7f8587e4e002 - __libc_start_main
  61:     0x559718f9a19e - _start
  62:                0x0 - <unknown>
[wayland-client error] A handler for wl_pointer panicked.
Aborted (core dumped)

```
**v0.6.x** panic message looks quite similar as in alacritty:
```
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/examples/kbd_input`
Window states: []
Window states: []
Gained focus while 0 keys pressed.
Modifiers changed ModifiersState { ctrl: false, alt: false, shift: false, caps_lock: false, logo: false, num_lock: true }
Window states: [Activated]
Key Pressed: 74.
 -> Received text "t".
Key Released: 74.
Key Pressed: 65.
 -> Received text "e".
Key Released: 65.
Key Pressed: 73.
 -> Received text "s".
Key Pressed: 74.
 -> Received text "t".
Key Released: 73.
Key Released: 74.
thread 'main' panicked at 'Memory allocation failure while loading a theme.', /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/cursor.rs:80:5
stack backtrace:
   0:     0x5560afc399f4 - backtrace::backtrace::libunwind::trace::h90669f559fb267f0
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1:     0x5560afc399f4 - backtrace::backtrace::trace_unsynchronized::hffde4e353d8f2f9a
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2:     0x5560afc399f4 - std::sys_common::backtrace::_print_fmt::heaf44068b7eaaa6a
                               at src/libstd/sys_common/backtrace.rs:77
   3:     0x5560afc399f4 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h88671019cf081de2
                               at src/libstd/sys_common/backtrace.rs:59
   4:     0x5560afc5307c - core::fmt::write::h4e6a29ee6319c9fd
                               at src/libcore/fmt/mod.rs:1052
   5:     0x5560afc370d7 - std::io::Write::write_fmt::hf06b1c86d898d7d6
                               at src/libstd/io/mod.rs:1426
   6:     0x5560afc3bb75 - std::sys_common::backtrace::_print::h404ff5f2b50cae09
                               at src/libstd/sys_common/backtrace.rs:62
   7:     0x5560afc3bb75 - std::sys_common::backtrace::print::hcc4377f1f882322e
                               at src/libstd/sys_common/backtrace.rs:49
   8:     0x5560afc3bb75 - std::panicking::default_hook::{{closure}}::hc172eff6f35b7f39
                               at src/libstd/panicking.rs:204
   9:     0x5560afc3b861 - std::panicking::default_hook::h7a68887d113f8029
                               at src/libstd/panicking.rs:224
  10:     0x5560afc3c1da - std::panicking::rust_panic_with_hook::hb7ad5693188bdb00
                               at src/libstd/panicking.rs:472
  11:     0x5560afb52ccb - std::panicking::begin_panic::h3b25ddf6e386753b
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panicking.rs:399
  12:     0x5560afb0903a - wayland_client::cursor::load_theme::hf7276cb0bed6f419
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/cursor.rs:80
  13:     0x5560afac69a9 - smithay_client_toolkit::pointer::theme::ScaledThemeList::get_cursor::hd5810c2a93e4b845
                               at src/pointer/theme.rs:46
  14:     0x5560afac6ead - smithay_client_toolkit::pointer::theme::ThemedPointer::set_cursor_with_scale::hff8666f5eb0d3cc9
                               at src/pointer/theme.rs:215
  15:     0x5560afad6f99 - smithay_client_toolkit::pointer::AutoPointer::set_cursor_with_scale::h354baddea9d03cc1
                               at src/pointer/mod.rs:136
  16:     0x5560afa91694 - smithay_client_toolkit::window::concept_frame::change_pointer::h4677d39e494f300e
                               at src/window/concept_frame.rs:784
  17:     0x5560afa81e1c - <smithay_client_toolkit::window::concept_frame::ConceptFrame as smithay_client_toolkit::window::Frame>::new_seat::{{closure}}::hfd5e96150329824d
                               at src/window/concept_frame.rs:321
  18:     0x5560afa87221 - smithay_client_toolkit::pointer::AutoThemer::theme_pointer_with_impl::{{closure}}::h9358699f23bad001
                               at src/pointer/mod.rs:69
  19:     0x5560afac0320 - smithay_client_toolkit::pointer::theme::ThemeManager::theme_pointer_with_impl::{{closure}}::{{closure}}::h828063cf9b7d6239
                               at src/pointer/theme.rs:143
  20:     0x5560afa7f8c8 - <alloc::boxed::Box<F> as core::ops::function::FnMut<A>>::call_mut::h2dadc443b5c43ed6
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/boxed.rs:1022
  21:     0x5560afababea - wayland_client::imp::proxy::proxy_dispatcher::{{closure}}::h93fff21d74b829f8
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/native_lib/proxy.rs:369
  22:     0x5560afaed289 - std::panicking::try::do_call::hde4234dd7a3f4ba6
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panicking.rs:305
  23:     0x5560afc3f727 - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:86
  24:     0x5560afaecaab - std::panicking::try::he574cc20b26c4d04
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panicking.rs:281
  25:     0x5560afaeb011 - std::panic::catch_unwind::h257cd54f2d167f56
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panic.rs:394
  26:     0x5560afab8b9a - wayland_client::imp::proxy::proxy_dispatcher::hd2f1c546aa39a041
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/native_lib/proxy.rs:353
  27:     0x7f36c8bf3f6b - <unknown>
  28:     0x7f36c8bf0662 - <unknown>
  29:     0x7f36c8bf1ccc - wl_display_dispatch_queue_pending
  30:     0x5560afb1891a - wayland_client::imp::event_queue::EventQueueInner::dispatch::h7a72f56ed5156b8f
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/native_lib/event_queue.rs:33
  31:     0x5560afb0b7b3 - wayland_client::event_queue::EventQueue::dispatch::h7dba2c64868bb2e9
                               at /home/ab/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/event_queue.rs:130
  32:     0x5560af9f5d59 - kbd_input::main::h79841bc06e5f4ae6
                               at examples/kbd_input.rs:141
  33:     0x5560afa24830 - std::rt::lang_start::{{closure}}::h8eb026a8b78a60af
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  34:     0x5560afc3bca3 - std::rt::lang_start_internal::{{closure}}::hb26e39676675046f
                               at src/libstd/rt.rs:52
  35:     0x5560afc3bca3 - std::panicking::try::do_call::he4701ab6e48d80c0
                               at src/libstd/panicking.rs:305
  36:     0x5560afc3f727 - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:86
  37:     0x5560afc3c680 - std::panicking::try::hd3de25f3cb7024b8
                               at src/libstd/panicking.rs:281
  38:     0x5560afc3c680 - std::panic::catch_unwind::h86c02743a24e3d92
                               at src/libstd/panic.rs:394
  39:     0x5560afc3c680 - std::rt::lang_start_internal::h9cf8802361ad86c2
                               at src/libstd/rt.rs:51
  40:     0x5560afa24809 - std::rt::lang_start::h784ad006a26a92c0
                               at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  41:     0x5560af9f6e9a - main
  42:     0x7f36c89a3002 - __libc_start_main
  43:     0x5560af9ec19e - _start
  44:                0x0 - <unknown>
[wayland-client error] A handler for wl_pointer panicked.
Aborted (core dumped)
```

`XCURSOR_SIZE=24 alacritty` works but with some window glitches like this:
![altty](https://user-images.githubusercontent.com/4666566/83137979-c5d7b780-a0f2-11ea-9333-ed3ba786f068.png)

**Comment by kchibisov** [comment]
>XCURSOR_SIZE=24 alacritty works but with some window glitches:

This has nothing to do with that glitch and it's likely a KDE (Wayland) bug, what's your XCURSOR_SIZE env variable? Just do `env | grep -i XCURSOR_SIZE`

**Comment by house-of-vanity** [comment]
> > XCURSOR_SIZE=24 alacritty works but with some window glitches:
> 
> This has nothing to do with that glitch and it's likely a KDE (Wayland) bug, what's your XCURSOR_SIZE env variable? Just do `env | grep -i XCURSOR_SIZE`

By default it's set to 0.

**Comment by kchibisov** [comment]
>By default it's set to 0.

Yeah, that's the problem, it shouldn't be zero, so that's why libwayland-client is failing. We can filter out that 0, but I guess you can figure out where it's set to 0 and change it to something else, or start with other xcursor size.

As for this space between decorations it's a known KDE bug(I'd assume so, since last time we've checked it was their bug) related to subsurface handling.

**Comment by kchibisov** [comment]
I've changed the title to reflect original problem, since that's something that should be fixed upstream (I've already send a patch for that, but I have no idea when it'll reach us).

The other issue you have should be brought to KDE folks.

**Comment by house-of-vanity** [comment]
Thanks for help!

**Comment by kchibisov** [comment]
Also, if `XCURSOR_SiZE=0` isn't something you set by yourself or somewhere in system, bring this to KDE as well.

**Comment by kchibisov** [comment]
The gap on startup and `XCURSOR` bug should be fixed in #4275, if you can test this branch on KWIN Wayland it'll be nice, since I only made some basic testing when adding work around for that gap bug.
